pkgs/by-name/no/noxdir/package.nix

```
@@ -0,0 +1,30 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "noxdir";
+  version = "0.8.0";
+
+  src = fetchFromGitHub {
+    owner = "crumbyte";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-fy/ETTGJKgVubf5s+Lgc0rW7q571mMjVh7V7/VBrRD4=";
+  };
+
+  vendorHash = "sha256-NtrTLF6J+4n+bnVsfs+WAmlYeL0ElJzwaiK1sk59z9k=";
+
+  checkPhase = "go test -v -buildvcs -race ./...";
```

phases must remain adjustable
```suggestion
  checkPhase = ''
    runHook preCheck
    go test -v -buildvcs -race ./...
    runHook postCheck
  '';
```